app/screens/ScheduleScreen/ScheduleScreen.tsx

```diff
@@ -11,63 +11,115 @@ import ScheduleCard, { ScheduleCardProps, Variants } from "./ScheduleCard"
 import { useStores } from "../../models"
 import { formatDate } from "../../utils/formatDate"
 import { useAppNavigation } from "../../hooks"
+import Animated, {
+  useAnimatedScrollHandler,
+  useSharedValue,
+  runOnJS,
+} from "react-native-reanimated"
+
+const { width } = Dimensions.get("window")
 
 export const ScheduleScreen: FC<TabScreenProps<"Schedule">> = observer(function ScheduleScreen() {
   useHeader({ title: "Schedule" })
   const navigation = useAppNavigation()
   const { schedulesStore } = useStores()
-  const { fetchData, selectedSchedule } = schedulesStore
+  const { fetchData, setSelectedSchedule, selectedSchedule, schedules } = schedulesStore
 
   useLayoutEffect(() => {
     fetchData()
   }, [fetchData])
 
+  const updateSchedule = (index) => setSelectedSchedule(schedules[index])
+
+  const scrollX = useSharedValue(0)
+  const ref = React.useRef(null)
+  const scrollHandler = useAnimatedScrollHandler({
```

Comment:
Add a `scrollHandler` so we can get get scroll X progress while dragging through the horizontal list